crates/models/llama/src/lib.rs

```diff
@@ -52,39 +63,54 @@ impl KnownModel for Llama {
 
         // model-global weights
         let wte = tl.load("tok_embeddings.weight")?;
-        let norm = tl.load("norm.weight")?;
-        let output = tl.load("output.weight")?;
+
+        let backend = if params.should_offload(0) {
```

Comment:
Seems like we should have a more explicit way of checking this. Maybe a `params.backend()` that returns the correct backend?